sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -463,23 +469,24 @@ public Coder<T> getOutputCoder() {
   static class UnboundedJmsReader<T> extends UnboundedReader<T> {
 
     private UnboundedJmsSource<T> source;
-    private JmsCheckpointMark checkpointMark;
     private Connection connection;
     private Session session;
     private MessageConsumer consumer;
     private AutoScaler autoScaler;
 
     private T currentMessage;
+    private Message currentJmsMessage;
     private Instant currentTimestamp;
 
-    public UnboundedJmsReader(UnboundedJmsSource<T> source, JmsCheckpointMark checkpointMark) {
+    Set<Message> messagesToAck;
```

Comment:
Keeping messages in memory seems like it could cause problems. Can we make this less memory intensive?

Reply:
Hello, 
The messagesToAck set is cleared every time we checkpoint (in the getCheckpointmark method).
Before that the same set was maintained in the JmsCheckpointmark. I don't think maintaining this is memory intensive.
What do you think?

Reply:
Yes it will work as the previous solution did